libs/isograph-react/src/core/cache.ts

```diff
@@ -262,13 +262,12 @@ function callSubscriptions(
               },
             );
 
-            if (
-              !areEqualObjectsWithDeepComparison(
-                subscription.readerAst,
-                subscription.encounteredDataAndRecords.item,
-                newEncounteredDataAndRecords.item,
-              )
-            ) {
+            const mergedItem = mergeObjectsUsingReaderAst(
```

Comment:
I think we need to replace `newEncounteredDataAndRecords.item` with `mergedItem`, e.g. on lines 276 and 280. What do you think?

Reply:
Wait, we don't. We just mutate the new object instead of returning an entirely new object. Okay, awesome, let's land